packages/compiler/src/server/client-config-provider.ts

```diff
@@ -0,0 +1,69 @@
+import { Connection } from "vscode-languageserver/node.js";
+import { ServerHost } from "./types.js";
+
+interface LSPConfig {
+  emit?: string[];
+}
+
+interface Config {
+  lsp?: LSPConfig;
```

Comment:
there is also technically the server-path that we define(despite this not being relevant to this it would be good to define)